packages/website/src/plasma/routes/input/pages/TextInputExamples.tsx

```diff
@@ -1,105 +1,106 @@
-    } else {
-        return {status: 'valid'};
-    }
-};
+const code = `
+    import * as React from 'react';
+    import {FormProvider, InputValidator, TextInput} from "@coveord/plasma-react";
 
-// start-print
-export const TextInputExamples: React.FunctionComponent = () => (
-    <VaporComponent id="TextInput" title="Text Input" withSource>
+    const nonEmptyValidation: InputValidator = (value: string) => {
+        const isEmpty = !value;
+        const isWhiteSpace = !isEmpty && !value.trim();
+        if (isWhiteSpace) {
+            return {status: 'warning', message: 'A white space is not empty, but is not ideal.'};
+        } else if (isEmpty) {
+            return {status: 'invalid', message: 'Cannot be empty.'};
+        } else {
+            return {status: 'valid'};
+        }
+    };
+
+    export default () => (
         <FormProvider>
-            <Section>
-                <TextInput
-                    required
-                    showValidationOnBlur
-                    validate={nonEmptyValidation}
-                    type="text"
-                    label="Label"
-                    title="Title"
-                    description="Description"
-                    helpText="Help text"
-                    tooltip="Tooltip"
-                />
-                <Section level={2} title="Disabled">
-                    <TextInput type="email" label="Email" disabled />
-                    <TextInput type="email" label="Email" defaultValue="123@abc.com" disabled />
-                </Section>
-                <Section level={2} title="useTextInput hook usage demo">
-                    <div className="flex space-around">
-                        <TextInput
-                            id="favorite-dish"
-                            type="text"
-                            label="Favorite Dish"
-                            defaultValue="pizza"
-                            validate={(value) =>
-                                /pizza/i.test(value)
-                                    ? {status: 'valid'}
-                                    : {status: 'warning', message: "Pizza is best but that's okay."}
-                            }
-                            showValidationOnBlur
-                        />
-                        <TextInput
-                            id="favorite-animal"
-                            type="text"
-                            label="Favorite Animal"
-                            defaultValue="Tiger"
-                            required
-                            validate={nonEmptyValidation}
-                            showValidationOnBlur
-                        />
-                    </div>
-                    <HookUsageDemo />
-                </Section>
-            </Section>
+            <TextInput
+                required
+                showValidationOnBlur
+                validate={nonEmptyValidation}
+                type="text"
```

Comment:
The type `text` could not be a default value?

Reply:
When implementing the TextInput, I thought about this and decided not to give it a default value. The reason is that people will not think about it and leave it as `text` for emails or urls, etc. Making it required forces us to think about which type it should have.